tgui/packages/tgui/interfaces/CrewConsole.tsx

```diff
@@ -0,0 +1,307 @@
+  }
+  if (jobId >= 200 && jobId < 230) {
+    return COLORS.department.centcom;
+  }
+  return COLORS.department.other;
+};
+
+const statToIcon = (life_status: number) => {
+  switch (life_status) {
+    case STAT_LIVING:
+      return 'heart';
+    case STAT_DEAD:
+      return 'skull';
+  }
+  return 'heartbeat';
+};
+
+const healthSort = (a: CrewSensor, b: CrewSensor) => {
+  if (a.life_status < b.life_status) return -1;
+  if (a.life_status > b.life_status) return 1;
+  if (a.health > b.health) return -1;
+  if (a.health < b.health) return 1;
+  return 0;
+};
+
+const areaSort = (a: CrewSensor, b: CrewSensor) => {
+  a.area ??= '~';
+  b.area ??= '~';
+  if (a.area < b.area) return -1;
+  if (a.area > b.area) return 1;
+  return 0;
+};
+
+const healthToAttribute = (
+  oxy: number,
+  tox: number,
+  burn: number,
+  brute: number,
+  attributeList: string[],
+) => {
+  const healthSum = oxy + tox + burn + brute;
+  const level = Math.min(Math.max(Math.ceil(healthSum / 25), 0), 5);
+  return attributeList[level];
+};
+
+type HealthStatProps = {
+  type: string;
+  value: number;
+};
+
+const HealthStat = (props: HealthStatProps) => {
+  const { type, value } = props;
+  return (
+    <Box inline width={2} color={COLORS.damageType[type]} textAlign="center">
+      {value}
+    </Box>
+  );
+};
+
+export const CrewConsole = () => {
+  return (
+    <Window title="Crew Monitor" width={600} height={600}>
+      <Window.Content scrollable>
+        <Section minHeight="540px">
+          <CrewTable />
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+type CrewSensor = {
+  name: string;
+  assignment: string | undefined;
+  ijob: number;
+  life_status: number;
+  oxydam: number;
+  toxdam: number;
+  burndam: number;
+  brutedam: number;
+  area: string | undefined;
+  health: number;
+  can_track: BooleanLike;
+  ref: string;
+};
+
+type CrewConsoleData = {
+  sensors: CrewSensor[];
+  link_allowed: BooleanLike;
+};
+
+const CrewTable = () => {
+  const { data } = useBackend<CrewConsoleData>();
+  const { sensors } = data;
+
+  const [sortAsc, setSortAsc] = useState(true);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [sortBy, setSortBy] = useState(SORT_OPTIONS[0]);
+
+  const cycleSortBy = () => {
+    let idx = SORT_OPTIONS.indexOf(sortBy) + 1;
+    if (idx === SORT_OPTIONS.length) idx = 0;
+    setSortBy(SORT_OPTIONS[idx]);
+  };
+
+  const nameSearch = createSearch(searchQuery, (crew: CrewSensor) => crew.name);
+
+  const sorted = sensors.filter(nameSearch).sort((a, b) => {
+    switch (sortBy) {
+      case 'name':
+        return sortAsc ? +(a.name > b.name) : +(b.name > a.name);
+      case 'ijob':
+        return sortAsc ? a.ijob - b.ijob : b.ijob - a.ijob;
+      case 'health':
+        return sortAsc ? healthSort(a, b) : healthSort(b, a);
+      case 'area':
+        return sortAsc ? areaSort(a, b) : areaSort(b, a);
+      default:
+        return 0;
+    }
+  });
+
+  const sortNames = {
```

Comment:
Should make this guy global too. Anything that doesn't touch with component state should move up in scope (including objects). My opinion is based off of this:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/HEAD/docs/rules/consistent-function-scoping.md